sdk/eventhub/Azure.Messaging.EventHubs.Processor/samples/Sample10_RunningInAzureStack.cs

```diff
@@ -0,0 +1,202 @@
+using Azure.Messaging.EventHubs.Consumer;
+using Azure.Messaging.EventHubs.Processor.Samples.Infrastructure;
+using Azure.Messaging.EventHubs.Producer;
+using Azure.Storage.Blobs;
+
+namespace Azure.Messaging.EventHubs.Processor.Samples
+{
+    /// <summary>
+    ///   An example of running the Event Processor in the Azure Stack platform.
+    /// </summary>
+    ///
+    public class Sample10_RunningInAzureStack : IEventHubsBlobCheckpointSample
+    {
+        /// <summary>
+        ///   The name of the sample.
+        /// </summary>
+        ///
+        public string Name => nameof(Sample10_RunningInAzureStack);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description => "An example of running the Event Processor in the Azure Stack platform.";
+
+        /// <summary>
+        ///   Runs the sample using the specified Event Hubs and Azure storage connection information.
+        /// </summary>
+        ///
+        /// <param name="eventHubsConnectionString">The connection string for the Event Hubs namespace that the sample should target.</param>
+        /// <param name="eventHubName">The name of the Event Hub, sometimes known as its path, that the sample should run against.</param>
+        /// <param name="blobStorageConnectionString">The connection string for the storage account where checkpoints and state should be persisted.</param>
+        /// <param name="blobContainerName">The name of the blob storage container where checkpoints and state should be persisted.</param>
+        ///
+        public async Task RunAsync(string eventHubsConnectionString,
+                                   string eventHubName,
+                                   string blobStorageConnectionString,
+                                   string blobContainerName)
+        {
+            // The latest Azure Storage API supported with Azure Stack Hub version 2002 is not compatible with the one that the Azure
+            // Storage SDK officially supports.  For this reason, the Event Processor Client, which makes use of the Azure Storage
+            // Blobs SDK, cannot be used with Azure Stack resources by default.  This sample illustrates how to work around this issue
+            // by using a pipeline policy to force the Azure Storage SDK to downgrade its version.
+            //
+            // In order to do this, create a Blob Client Options instance and add a Downgrade Version Policy to it.  These options
+            // will be passed to the Blob Container Client used by the processor.  The implementation of the policy is provided at
+            // the end of this sample.
+
+            string consumerGroup = EventHubConsumerClient.DefaultConsumerGroupName;
+
+            BlobClientOptions storageClientOptions = new BlobClientOptions();
+            storageClientOptions.AddPolicy(new DowngradeVersionPolicy(), HttpPipelinePosition.PerCall);
```

Comment:
Does DowngradeVersionPolicy support specifying any (old or current) version of the Storage service? if so, then reconsider renaming that to StorageAPIVersionPolicy or the like. At least in Java, you do not name classes starting with a verb. They should be nouns. The action happens in the methods :)

Reply:
Renamed it.